packages/settings/src/settings.ts

```diff
@@ -59,13 +50,77 @@ export class Settings {
    *
    */
   async save(plugin: string, raw: string): Promise<void> {
-    await this._storage.setItem(plugin, raw);
+    await Private._storage.setItem(plugin, raw);
   }
+}
 
-  private _storage = localforage.createInstance({
+namespace Private {
+  let _all: { settings: IPlugin[] };
+  const _federated = new Map<string, IPlugin | undefined>();
+
+  export const _storage = localforage.createInstance({
     name: STORAGE_NAME,
     description: 'Offline Storage for Settings',
     storeName: 'settings',
     version: 1
   });
+
+  export async function getAll(): Promise<{ settings: IPlugin[] }> {
```

Comment:
Looks like the settings are not updated on the RTD preview for this PR, for example when switching the theme or moving tabs to the left and right area: https://jupyterlite--58.org.readthedocs.build/en/58/_static/lab/index.html

Although refreshing the page does pick them up. So maybe it's related to the new `getAll` "merging" logic.

Reply:
yeah, gross. needs test #6 lol

Reply:
Yeah absolutely.

Just pushed c0bff5fe00cd05ad693f32d6b9c40d9503502ed9 in the meantime which should fix it.

It adds the network requests back to fetch `all.json` and the federated extension schemas too. But it would fine to take a look at it again in a separate PR if we want to optimize this.

Reply:
This should also handle editing settings for a federated extension via the settings editor.

Reply:
yeah, settings editing works now that the `isFederated` check is accurate